lib/flagutil/duration.go

```diff
@@ -10,45 +10,47 @@ import (
 	"github.com/VictoriaMetrics/metricsql"
 )
 
-// NewDuration returns new `duration` flag with the given name, defaultValue and description.
+// NewRetentionDuration returns new `duration` flag with the given name, defaultValue and description.
 //
 // DefaultValue is in months.
-func NewDuration(name string, defaultValue string, description string) *Duration {
+func NewRetentionDuration(name string, defaultValue string, description string) *RetentionDuration {
 	description += "\nThe following optional suffixes are supported: s (second), h (hour), d (day), w (week), y (year). " +
 		"If suffix isn't set, then the duration is counted in months"
-	d := &Duration{}
+	d := &RetentionDuration{}
 	if err := d.Set(defaultValue); err != nil {
 		panic(fmt.Sprintf("BUG: can not parse default value %s for flag %s", defaultValue, name))
 	}
 	flag.Var(d, name, description)
 	return d
 }
 
-// Duration is a flag for holding duration.
-type Duration struct {
+// RetentionDuration is a flag for holding duration for retention period.
+type RetentionDuration struct {
 	// msecs contains parsed duration in milliseconds.
 	msecs int64
 
 	valueString string
 }
 
 // Duration returns d as time.Duration
-func (d *Duration) Duration() time.Duration {
+func (d *RetentionDuration) Duration() time.Duration {
 	return time.Millisecond * time.Duration(d.msecs)
 }
 
 // Milliseconds returns d in milliseconds
-func (d *Duration) Milliseconds() int64 {
+func (d *RetentionDuration) Milliseconds() int64 {
 	return d.msecs
 }
 
 // String implements flag.Value interface
-func (d *Duration) String() string {
+func (d *RetentionDuration) String() string {
 	return d.valueString
 }
 
 // Set implements flag.Value interface
-func (d *Duration) Set(value string) error {
+// It assumes that value without unit should be parsed as `month` duration.
+// It returns an error iv value has `m` unit.
```

Comment:
```suggestion
// It returns an error if value has `m` unit.
```
:)

Reply:
fixed in https://github.com/VictoriaMetrics/VictoriaMetrics/commit/aa6c237603ba4bfc36929ff0872ce9b6e9aaf4b5